File: discover_ruby/github_most_popular.rb
# Send a request to Github's API to retrieve JSON data for 30 most popular Ruby project

require 'httpclient'
require 'json'

extheaders = {
  'User-Agent' => 'Holberton_School',
  'Authorization' => 'token 3d59bbfb240064a050cb72476c091bd363e5267c'
}

http = HTTPClient.new
response = http.get_content("https://api.github.com/search/repositories?q=language:ruby&sort=stars&order=desc", nil, extheaders)
puts(response)

File: discover_more_python/github_most_popular.py
""" This program prints the response from Github's API when requesting 
    the 30 most popular Python projects
"""
from urllib2 import Request, urlopen, URLError

request_headers = {
            'User-Agent': 'Holberton School',
            'Authorization': 'token ae6b5eaf4971c9e50adafdb86513c15df7de6fbb'
        }

request = Request('https://api.github.com/search/repositories?q=language:python&sort=stars&order=desc', headers=request_headers)

try:
        response = urlopen(request)
        string = response.read()
        print string
except URLError, e:
    print "Error code: ", e
